### PR TITLE
Better integration of UndefinedClasses in CodeImporter

### LINKED ISSUE #14967: ShiftClassBuilder and UndefinedClass should be better integrated.
Right now the resolver (a strategy responsible to check the superclass does not exist) is only invoked from the ClassDefinitionParser!!!!?

```
ShiftClassBuilder >> classNamed: aName

	^ self buildEnvironment at: aName ifAbsent: [ nil ]
```

should invoke the resolver.

We should pay attention that classNamed: is also invoked on the class.

```
tryToFillOldClass

	oldClass ifNotNil: [ ^ self ].
	name ifNil: [ ^ self ].
	self oldClass: (self classNamed: self name)
```

### PR DESCRIPTION
**Pull request by Ducasse**
@jecisc please have a look

### CODE REVIEW

**Comment by MarcusDenker** [comment]
When doing the Full Image Build, we get a problem:

```
+ /builds/workspace/est_and_branch_Pipeline_PR-14972/64/bootstrap-downloads/vmtarget/pharo --headless Pharo12.0-PR-monticello_bootstrap-64bit-d0b1a7e.image --no-default-preferences st /builds/workspace/est_and_branch_Pipeline_PR-14972/64/bootstrap-cache/st-cache/Monticello.st --save --quit
Instance of CodeImporter did not understand #enableUndefinedClass
CodeImporter(Object)>>doesNotUnderstand: #enableUndefinedClass
buildClassFromOldDefinition: t1
  self error: 'No decompiler available' in DoItChunk>>buildClassFromOldDefinition: in Block: buildClassFromOldDefinition: t1...
ShiftClassInstaller>>make:
ShiftClassInstaller class>>make:
DoItChunk>>buildClassFromOldDefinition:
DoItChunk>>importFor:logSource:
evaluate
```

**Comment by jecisc** [comment]
@Ducasse What should be the default value of enableUndefinedClass in CodeImporter?

**Comment by Ducasse** [comment]
is it false so that we do not create UndefinedClass

**Comment by jecisc** [comment]
I had this exacte same bug on another PR about the code importer and the problem is that the environment of the installed class is wrong and then during the lookup of the method, the method is not found because it's looked in the wrong class

File: src/CodeImport/DoItChunk.class.st [hunk 1]
```diff
@@ -150,11 +153,14 @@ DoItChunk >> buildClassFromOldDefinition: ast [
...
-		  aBuilder
-			  superclass: superclass;
-			  environment: superclass environment;
+		  
+			importer enableUndefinedClass 
+				ifTrue: [ aBuilder useUndefinedClass ]
+				ifFalse: [ aBuilder useStrictSuperclass ].
+		  
+		aBuilder
+			  superclassName: superclassName;
```
Comment (guillep) [inline src/CodeImport/DoItChunk.class.st]:
```suggestion
			  superclassName: superclassName asSymbol;
```

**Comment by MarcusDenker** [comment]
this was merged in Pharo12. Is that correct?

**Comment by MarcusDenker** [comment]
ah, no, closed

**Pull request closed without merging**